internal/command/arguments/extended.go

```diff
@@ -128,28 +133,80 @@ func parseTargetables(rawTargetables []string, flag string) ([]addrs.Targetable,
 	return targetables, diags
 }
 
-func parseRawTargetsAndExcludes(targets []string, excludes []string) ([]addrs.Targetable, []addrs.Targetable, tfdiags.Diagnostics) {
-	var parsedTargets []addrs.Targetable
-	var parsedExcludes []addrs.Targetable
+// parseFile gets a filePath and reads the file, which contains a list of targets
+// with each line in the file representating a targeted object, and returns
+// a list of addrs.Targetable. This is used for parsing the input of -target-file
+// and -exclude-file flags
+func parseFileTargetables(filePaths []string, flag string) ([]addrs.Targetable, tfdiags.Diagnostics) {
+
+	// If no file passed, no targets
+	if len(filePaths) <= 0 {
+		return nil, nil
+	}
+	var targetables []addrs.Targetable
 	var diags tfdiags.Diagnostics
 
-	if len(targets) > 0 && len(excludes) > 0 {
+	for _, filePath := range filePaths {
+		b, err := os.ReadFile(filePath)
+		diags = diags.Append(err)
```

Comment:
If the read returns an error, we probably don't want to even attempt scanning the file and can continue to the next loop iteration.